src/virus_clade_utils/util/sequence.py

```diff
@@ -54,6 +62,74 @@ def get_covid_genome_data(released_since_date: str, base_url: str, filename: str
     logger.info("NCBI API call completed", elapsed=elapsed)
 
 
+def download_covid_genome_metadata(url: str, data_path: Path) -> Path:
+    """Download the latest GenBank genome metadata data from Nextstrain."""
+
+    session = get_session()
+    filename = data_path / Path(url).name
+
+    start = time.perf_counter()
+    with session.get(url, stream=True) as result:
+        result.raise_for_status()
+        with open(filename, "wb") as f:
+            for chunk in result.iter_content(chunk_size=None):
+                f.write(chunk)
+
+    end = time.perf_counter()
+    elapsed = end - start
+    logger.info("genome metadata downloaded", elapsed=elapsed)
+
+    return result[0]
+
+
+def get_covid_genome_metadata(metadata_path: Path, num_rows: int | None = None) -> pl.LazyFrame:
+    """Read GenBank genome metadata into a Polars LazyFrame."""
+
+    if (compression_type := metadata_path.suffix) == ".zst":
+        metadata = pl.scan_csv(metadata_path, separator="\t", n_rows=num_rows)
+    elif compression_type == ".xz":
```

Comment:
The full metadata file that we need is .zst (ZSTD compression), which works fine with Polars `scan_csv` operation. 

However, some of the smaller metadata sample files use a different compression method (annoying!), so this function handles both.